### PR TITLE
cp-103: Add dropdown menu

### PR DESCRIPTION
**Pull request by Nanaier**
Demo: 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/90211108/121c39a0-f90a-4dfb-b190-4ea2e4e8fa00

### CODE REVIEW

**Review by OlesiaTr — DISMISSED** [review]

File: frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { useCallback, useLocation, useState } from '#libs/hooks/hooks.js';
+import { type Routes } from '#libs/types/types.js';
+
+import styles from './styles.module.scss';
+
+const Dropdown: React.FC<Routes> = ({ routes }) => {
+  const { pathname } = useLocation();
+  const [isOpen, setOpen] = useState(false);
+
+  const toggleDropdown = useCallback((): void => {
```
Comment (v0ldemar01) [inline frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx]: Please follow rule about handler's naming

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx (hunk 1 above)
Comment (Nanaier) [reply frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx]: done!

File: frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { Link } from '#libs/components/components.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useLocation, useState } from '#libs/hooks/hooks.js';
+import { type Routes } from '#libs/types/types.js';
+
+import styles from './styles.module.scss';
+
+const Dropdown: React.FC<Routes> = ({ routes }) => {
```
Comment (liza-veis) [inline frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx]: Component name should be the same as file name
```suggestion
const DropdownMenu: React.FC<Routes> = ({ routes }) => {
```

File: frontend/src/libs/components/dropdown-menu/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    brightness(94%) contrast(97%);
+}
+
+.title {
+  display: flex;
+  justify-content: baseline;
+  align-self: last baseline;
+}
+
+@media (width > 480px) {
```
Comment (liza-veis) [inline frontend/src/libs/components/dropdown-menu/styles.module.scss]: let's write all queries in the same style
```suggestion
@media (width < 480px) {
```

File: frontend/src/libs/components/header/styles.module.scss [hunk 4]
```diff
@@ -18,3 +18,9 @@
   display: flex;
   align-items: center;
 }
+
+@media (width <= 480px) {
```
Comment (liza-veis) [inline frontend/src/libs/components/header/styles.module.scss]:
```suggestion
@media (width < 480px) {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/libs/types/route/route.type.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { type AppRoute } from '#libs/enums/enums.js';
+
+import { type ValueOf } from '../types.js';
```
Comment (fshabanov) [inline frontend/src/libs/types/route/route.type.ts]: This will cause a circular dependency

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Comment by Nanaier** [comment]
@Liza-Veis hey, as you can see I have some conflicts rn - Would it be okay if I resolve them based on the clean-up of the code done by my other teammate? So there would be no code showcasing how the dropdown works on the root page since it would be cleaned up, but you can see the functionality on the demo

**Review by Ekit18 — DISMISSED** [review]

File: frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+type Properties = {
+  routes: Route[];
+};
+
+const DropdownMenu: React.FC<Properties> = ({ routes }) => {
+  const { pathname } = useLocation();
+  const [isOpen, setOpen] = useState(false);
+
+  const handleDropdownToggle = useCallback((): void => {
+    setOpen(!isOpen);
```
Comment (Ekit18) [inline frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx]: may be its better to write smth like this: setOpen((previous)=> !previous)

File: frontend/src/libs/components/dropdown-menu/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+.item {
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-self: baseline;
+  gap: 5px;
+}
+
+hr {
```
Comment (dmytro-kucherenko) [inline frontend/src/libs/components/dropdown-menu/styles.module.scss]: According to the rule B10 we can't use tag names in styles, only classes

File: frontend/src/libs/components/sidebar/styles.module.scss [hunk 8]
```diff
@@ -64,3 +64,9 @@
 .icons-container a {
```
Comment (dmytro-kucherenko) [inline frontend/src/libs/components/sidebar/styles.module.scss]: The same with tag names

**Review by dmytro-kucherenko — COMMENTED** [review]
Nice work)

**Review by Vika030718 — DISMISSED** [review]

File: frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+  return (
+    <div className={styles['dropdown']}>
+      <button
+        className={styles['dropdown-header']}
+        onClick={handleDropdownToggle}
+      >
+        <span className={styles['bar']} />
+        <span className={styles['bar']} />
+        <span className={styles['bar']} />
```
Comment (HIMDanny) [inline frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx]: according to the **B4** rule, we can reduce the number of spans and use pseudo-elements instead

**Review by HIMDanny — DISMISSED** [review]

**Review by OlenaYatseniuk — DISMISSED** [review]

**Review by liza-veis — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**